saleor/dashboard/staff/forms.py

```diff
@@ -44,3 +63,12 @@ def __init__(self, *args, **kwargs):
         if address.last_name:
             placeholder = get_name_placeholder(address.last_name)
             self.fields['last_name'].widget.attrs['placeholder'] = placeholder
+
+    def clean(self):
+        cleaned_data = super().clean()
+
+        # Remove all permissions if user is not staff
+        if not cleaned_data['is_staff']:
+            cleaned_data['user_permissions'] = []
```

Comment:
It would be nice to add tests for this logic.

Reply:
Sure, test is added